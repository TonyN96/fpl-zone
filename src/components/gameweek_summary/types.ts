export interface SummaryStatType {
  label: string;
  teamCode?: number;
  playerName?: string;
  value?: string | number;
}
