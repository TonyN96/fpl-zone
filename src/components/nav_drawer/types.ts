export interface MenuItemType {
  id: string;
  label: string;
  type: "nav" | "auth";
  onClick: () => void;
}
