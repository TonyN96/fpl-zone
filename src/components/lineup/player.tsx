import React from "react";
import { Box, Grid, Typography } from "@mui/material";
import { Player as PlayerType } from "types/player";

interface PlayerProps {
  player: PlayerType;
}

export default function Player({ player }: PlayerProps): JSX.Element {
  return (
    <Grid container direction="column" alignItems="center">
      <Grid item>
        <img
          src={`${process.env.PUBLIC_URL}/assets/images/kits/${player.team_code}.png`}
          alt="kit-img"
          height={50}
          width={50}
        />
      </Grid>
      <Grid item>
        <Typography>{player.web_name}</Typography>
      </Grid>
      <Grid item>
        <Typography>{player.event_points}</Typography>
      </Grid>
    </Grid>
  );
}
