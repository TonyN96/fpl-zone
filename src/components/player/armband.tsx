import React from "react";
import { Box, Typography } from "@mui/material";

interface ArmbandProps {
  isVice?: boolean;
}

export const Armband = ({ isVice = false }: ArmbandProps): JSX.Element => {
  return (
    <Box
      bgcolor='rgb(240, 240, 240, 1)'
      border='1px solid black'
      borderRadius='50%'
      className='flex-center'
      data-testid='armband-container'
      height='1.8vw'
      left={0}
      position='absolute'
      top={0}
      width='1.8vw'
    >
      <Typography>{isVice ? "V" : "C"}</Typography>
    </Box>
  );
};
