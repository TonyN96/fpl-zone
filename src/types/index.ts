export * from "./app_data";
export * from "./fixture";
export * from "./game_data";
export * from "./game_settings";
export * from "./gameweek";
export * from "./league";
export * from "./lineup";
export * from "./phase";
export * from "./player";
export * from "./player_past";
export * from "./player_performance";
export * from "./player_stat";
export * from "./player_summary";
export * from "./position";
export * from "./region";
export * from "./team";
export * from "./team_data";
export * from "./team_history";
export * from "./team_picks";
export * from "./transfer";
