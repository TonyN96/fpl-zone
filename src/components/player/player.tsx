import React, { useContext } from "react";
import { Box, Typography, useTheme } from "@mui/material";
import { AppDataContext } from "app_content";
import { getTeamKitImageUrl } from "helpers";
import { Player as PlayerType } from "types/player";

import { Armband } from "./armband";

import "./player.css";

interface PlayerProps {
  player: PlayerType;
  handlePlayerPerformanceClick: (player: PlayerType) => void;
  multiplier: number;
  isCaptain?: boolean;
  isViceCaptain?: boolean;
}

export default function Player({
  player,
  handlePlayerPerformanceClick,
  multiplier,
  isCaptain = false,
  isViceCaptain = false
}: PlayerProps): JSX.Element {
  const theme = useTheme();
  const { isMobile } = useContext(AppDataContext);
  const isGoalkeeper = player.element_type === 1;
  const url = getTeamKitImageUrl(player.team_code, isGoalkeeper);

  const Name = (): JSX.Element => (
    <Box
      bgcolor={theme.palette.primary.main}
      className='flex-center'
      height='100%'
      overflow='hidden'
      p='0.1rem'
      width='100%'
    >
      <Typography
        className='text-ellipsis'
        color={theme.palette.info.main}
        data-testid='player-name'
        fontSize={isMobile ? "0.8rem" : theme.typography.body1.fontSize}
      >
        {player.web_name.toUpperCase()}
      </Typography>
    </Box>
  );

  const Score = (): JSX.Element => (
    <Box
      bgcolor={theme.palette.secondary.main}
      className='flex-center'
      height='100%'
      overflow='hidden'
      p='0.1rem'
      width='100%'
    >
      <Typography
        data-testid='player-score'
        fontSize={isMobile ? "1rem" : theme.typography.body1.fontSize}
        px={1}
      >
        {player.event_points * multiplier}
      </Typography>
    </Box>
  );

  return (
    <Box
      className='flex-center'
      data-testid={`player-container-${player.id}`}
      flexDirection='column'
      height='100%'
      maxWidth='8rem'
      onClick={(): void => handlePlayerPerformanceClick(player)}
      overflow='hidden'
      position='relative'
      sx={{ cursor: "pointer" }}
      width='18%'
    >
      {isCaptain && <Armband />}
      {isViceCaptain && <Armband isVice />}
      <img
        alt='team-kit'
        className='kit-img'
        data-testid={`kit-img-player-${player.id}`}
        src={url}
      />
      <Box
        bottom={0}
        display='flex'
        flexDirection='column'
        justifyContent='center'
        position='absolute'
        width='100%'
      >
        <Name />
        <Score />
      </Box>
    </Box>
  );
}
