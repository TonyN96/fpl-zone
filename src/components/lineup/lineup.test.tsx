import React from "react";
import { render, screen } from "@testing-library/react";
import { mockAppData, mockPlayers, mockPositions } from "test";
import "@testing-library/jest-dom/extend-expect";
import Lineup from "components/lineup/lineup";
import { Player } from "types/player";
import { AppDataContext } from "app_content";
import "@testing-library/jest-dom/extend-expect";

describe("Lineup Tests", () => {
  const mockSelected: Player[][] = [];
  mockPositions.slice(0, 10).forEach((pos) => {
    const players: Player[] = [];
    mockPlayers.forEach((player) => {
      player.element_type === pos.id && players.push(player);
    });
    mockSelected.push(players);
  });

  const mockBench = mockPlayers.slice(11, 14) as Player[];

  const createComponent = (): JSX.Element => {
    return (
      <AppDataContext.Provider value={mockAppData}>
        <Lineup selected={mockSelected} bench={mockBench} />
      </AppDataContext.Provider>
    );
  };

  it("Snapshot test", () => {
    const { asFragment } = render(createComponent());
    expect(asFragment()).toMatchSnapshot();
  });
  describe("Renders lineup as expected", () => {
    it("First XI", () => {
      render(createComponent());
      const selectedGrid = screen.getByTestId("first-xi-players");
      mockSelected.forEach((positionGroup) => {
        positionGroup.forEach((player) => {
          expect(selectedGrid).toHaveTextContent(player.web_name);
          expect(selectedGrid).toHaveTextContent(player.event_points.toString());
        });
      });
    });
    it("Bench", () => {
      render(createComponent());
      const benchGrid = screen.getByTestId("bench-players");
      mockBench.forEach((player) => {
        expect(benchGrid).toHaveTextContent(player.web_name);
        expect(benchGrid).toHaveTextContent(player.event_points.toString());
      });
    });
  });
});
