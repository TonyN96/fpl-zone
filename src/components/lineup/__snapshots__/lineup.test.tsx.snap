// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lineup Tests Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-12kpaeh"
  >
    <div
      class="MuiBox-root css-1bxf5j8"
      data-testid="first-xi-players"
    >
      <div
        class="MuiBox-root css-geg32f"
      >
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="662"
        >
          <div
            class="MuiBox-root css-m3owix"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Triner
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  30
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="9"
        >
          <div
            class="MuiBox-root css-m3owix"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Bauldrey
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  23
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="550"
        >
          <div
            class="MuiBox-root css-1386e90"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Clampe
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  16
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-geg32f"
      >
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="828"
        >
          <div
            class="MuiBox-root css-1hwhk2i"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Jankin
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  13
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="324"
        >
          <div
            class="MuiBox-root css-ki4kbn"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Summerill
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  13
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="929"
        >
          <div
            class="MuiBox-root css-ramzw9"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Malloy
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  13
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="156"
        >
          <div
            class="MuiBox-root css-yvqdr8"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Turbefield
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  9
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="286"
        >
          <div
            class="MuiBox-root css-1386e90"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  MacMenemy
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  7
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="304"
        >
          <div
            class="MuiBox-root css-4x3s25"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Sives
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  4
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-geg32f"
      >
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="756"
        >
          <div
            class="MuiBox-root css-yvqdr8"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Miettinen
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  6
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="428"
        >
          <div
            class="MuiBox-root css-4x3s25"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Macek
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  7
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="938"
        >
          <div
            class="MuiBox-root css-ramzw9"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Perrington
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  16
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="505"
        >
          <div
            class="MuiBox-root css-1w92tz0"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Muller
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  24
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="137"
        >
          <div
            class="MuiBox-root css-1b560z9"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Tingley
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  12
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-geg32f"
      >
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="466"
        >
          <div
            class="MuiBox-root css-1b560z9"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Workes
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  22
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1alt9xp"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Bench
      </h5>
      <div
        class="MuiBox-root css-1doxykz"
        data-testid="bench-players"
      >
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="286"
        >
          <div
            class="MuiBox-root css-1386e90"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  MacMenemy
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  7
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="304"
        >
          <div
            class="MuiBox-root css-4x3s25"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Sives
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  4
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-hgxypz"
          data-testid="137"
        >
          <div
            class="MuiBox-root css-1b560z9"
            data-testid="kit-img-container"
          />
          <div
            class="MuiBox-root css-9pa3st"
          >
            <div
              class="MuiBox-root css-3e87oq"
            >
              <div
                class="MuiBox-root css-2f68vp"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-z1doim-MuiTypography-root"
                  data-testid="player-name"
                >
                  Tingley
                </p>
              </div>
              <div
                class="MuiBox-root css-13w0h3v"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  data-testid="player-score"
                >
                  12
                </p>
              </div>
              <div
                aria-label="Performance"
                class="MuiBox-root css-1bii6vr"
                data-mui-internal-clone-element="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
